source/Camera/Basic/C++Halcon/CaptureViaGenICam/CaptureViaGenICam.cpp

```diff
@@ -20,14 +20,10 @@ void setColorsInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const Ha
 	HalconCpp::HTuple rows, cols;
 	domain.GetRegionPoints(&rows, &cols);
 
-	const HalconCpp::HTuple Rtup = RGB.AccessChannel(1).GetGrayval(rows, cols);
-	const HalconCpp::HTuple Gtup = RGB.AccessChannel(2).GetGrayval(rows, cols);
-	const HalconCpp::HTuple Btup = RGB.AccessChannel(3).GetGrayval(rows, cols);
-
 	std::cout << "Adding RGB to ObjectModel3D" << std::endl;
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", RGB.AccessChannel(1).GetGrayval(rows, cols));
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", RGB.AccessChannel(2).GetGrayval(rows, cols));
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", RGB.AccessChannel(3).GetGrayval(rows, cols));
```

Comment:
Is this where you avoid one copy?

Reply:
I thought so, but it doesn't give noticable time difference. Maybe it creates the HTuple inside SetObjectModel3DAttribMod() function anyway